### DISCUSSION TITLE
Should we enforce 2FA for coordinator?

### DISCUSSION THREAD

**Discussion by baumstern** [quadratic-funding/mpc-phase2-suite#158] [Ideas]
reference: [Upgrade to Cloud Identity Platform for multi-factor authentication](https://firebase.google.com/support/guides/security-checklist#upgrade_to_cloud_identity_platform_for_multi-factor_authentication)

**Comment by baumstern** [comment]
@0xjei @ctrlc03

**Reply by 0xjei** [reply]
Thanks Dae for bringing this discussion to the table :+1: 

I hope to not misunderstanding the question! Currently, I'm wondering whether or not it makes sense to add this MFA mechanism in our MPC Tool since any coordinator is going to authenticate with Github. Therefore, since Github provide MFA mechanisms and to break our MPC Tool MFA, someone needs to authenticate using the Github account of the coordinator in the CLI. In order to do that, needs to log in the browser and break the MFA mechanisms which are active on the coordinator's Github. IMHO, this look like an extra layer of protection which is always good but a bit redundant if and only if the coordinator has other MFA mechanisms active on their Github account.

Did you have consider these aspects as well? Let me know what do you think about :thought_balloon:

**Reply by ctrlc03** [reply]
I think MFA on GitHub will suffice for interaction via the MPC tool. However, coordinators should follow security best practices (which we will provide) for setting up the required services, and MFA on their firebase account is definitely one, as the infrastructure might be the part most likely to be attacked.
